src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1724,40 +1724,50 @@
   {:pre [(or (not non-migrator-name)
              (and non-migrator-name db-name))]}
 
-  (when non-migrator-name
-    (jdbc/revoke-role-db-access non-migrator-name db-name))
-
-  (try
-    (when non-migrator-name
-      ;; Because the revoke may not actually produce an error when
-      ;; it doesn't work.
-      (when (jdbc/has-database-privilege? non-migrator-name db-name "connect")
-        (throw
-         (ex-info (str "Unable to prevent non-migrator connections during migration ")
-                  {:kind ::unable-to-block-other-pdbs-during-migration})))
-      (jdbc/disconnect-db-role db-name non-migrator-name))
-
-    (jdbc/with-db-transaction []
-      (require-schema-migrations-table)
-      (jdbc/do-commands
-       "lock table schema_migrations in access exclusive mode")
-      (require-valid-schema)
-      (let [tables (set/union (run-migrations (pending-migrations))
-                              (create-indexes))]
-        (note-migrations-finished)
-        tables))
-
-    (finally
-      (when non-migrator-name
-        ;; This won't run if the jvm doesn't quit in a way that lets
-        ;; threads finish (TK's shutdown process does).
-        (try
-          (jdbc/restore-role-db-access non-migrator-name db-name)
-          (catch Throwable ex
-            ;; Don't let this rethrow, so that it won't suppress any
-            ;; pending exception.
-            (log/error
-             ex (trs "Unable to restore db access after migrations"))))))))
+  ;; When a non-migrator-name is given, do our part to attempt to
+  ;; prevent concurrent migrations or any access to a database that's
+  ;; at an unexpected migration level.  See the connectionInitSql
+  ;; setting for related efforts.
+  (let [coordinate? non-migrator-name
+        orig-user (and coordinate? (jdbc/current-user))]
```

Comment:
Might be good to add a comment here about the current-user being the migrator user in this case because the migrations use a separate connection pool set up with that user and how any jdbc calls in the form will use the migrator user.